### PR TITLE
[Ehn] Enhance config module

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
### PR types

New features
### PR changes

APIs 
### Describe

1. 通过`ConfigStore`，支持在yaml文件中指定`defaults`字段来提供通用参数的默认配置，免去重复的yaml字段填写和`Solver`实例化时的十多个参数指定，如下所示。
    <table>
      <tr>
        <td style="vertical-align: top;">
          <pre style="vertical-align: top;"><code>
    # before 
    solver = ppsci.solver.Solver(
        model,
        constraint,
        cfg.output_dir,
        optimizer,
        lr_scheduler,
        cfg.TRAIN.epochs,
        cfg.TRAIN.iters_per_epoch,
        save_freq=cfg.TRAIN.save_freq,
        log_freq=cfg.log_freq,
        eval_during_train=True,
        eval_freq=cfg.TRAIN.eval_freq,
        seed=cfg.seed,
        equation=equation,
        geom=geom,
        validator=validator,
        visualizer=visualizer,
        pretrained_model_path=cfg.TRAIN.pretrained_model_path,
        checkpoint_path=cfg.TRAIN.checkpoint_path,
        eval_with_no_grad=cfg.EVAL.eval_with_no_grad,
    )</code></pre>
        </td>
        <td style="vertical-align: top;">
          <pre style="vertical-align: top;"><code>
    # after
    solver = ppsci.solver.Solver(
        model,
        constraint,
        optimizer=optimizer,
        equation=equation,
        validator=validator,
        visualizer=visualizer,
        cfg=cfg, <-- Only need cfg
    )</code></pre>
        </td>
      </tr>
    </table>
3. 移除原有的AttrDict相关代码。
4. 更新相关单测

此PR只负责更新基础功能，案例的代码、文档更新在 <https://github.com/PaddlePaddle/PaddleScience/pull/852>

### CODE REVIEW

**Review by zhiminzhang0830 — APPROVED** [review]
LGTM
